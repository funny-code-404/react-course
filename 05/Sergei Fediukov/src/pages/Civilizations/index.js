import React from 'react'
import { Navigation, IsLoading } from '../../components'
import CivilizationsContext from '../../context/CivilizationsContext/index'
import { useContext } from 'react'
import { StyledDiv, StyledButton } from '../../Styles/styles'

const Civilizations = (props) => {
    const handleRedirect = () => {
        props.history.push('/')
    }
    const { data } = useContext(CivilizationsContext)
    return (
        <div>
            <StyledButton onClick={handleRedirect}>go home</StyledButton>
            <h1>
                Civilizations:
            </h1>
            <StyledDiv>
                <>
                    {
                        data[0].start ? <IsLoading /> :
                            <Navigation menu={data} nav={props.location.pathname} />
                    }
                </>
            </StyledDiv>
        </div>
    )
}
export default Civilizations