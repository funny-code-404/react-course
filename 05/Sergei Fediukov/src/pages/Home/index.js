const Home = () => {
    return (
        <div>
            <h1>
                home page
           </h1>
        </div>
    )
}
export default Home