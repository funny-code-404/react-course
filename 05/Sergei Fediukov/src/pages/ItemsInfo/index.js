import React from 'react'
import IsLoading from '../../components/IsLoading'


class ItemsInfo extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            // details: this.props.details,
            flag: false,
        }
    }
    // static getDerivedStateFromProps(nextProps, prevState) {
    //     if (nextProps.details !== prevState.details) {
    //         return {
    //             details: nextProps.detals,
    //             flag: !prevState.flag
    //         }
    //     }
    // }
    componentDidUpdate(prevProps) {
        if (prevProps.details !== this.props.details) {
            this.setState({
                flag: !prevProps.flag
            })
        }
    }
    render() {
        const { name, army_type, expansion, team_bonus, civilization_bonus, unique_unit, unique_tech, description, age, created_in, cost, build_time, reload_time, attack_delay, movement_rate, line_of_sight, hit_points, range, attack, armor, accuracy, special, attack_bonus, applies_to, search_radius } = this.props.details
        console.log(this.props)
        const handleGoBack = () => {
            this.props.history.goBack()
        }
        const handleRedirect = () => {
            this.props.history.push('/')
        }

        return (
            <div>
                {
                    this.state.flag ?
                        <IsLoading /> :
                        <div className="details">
                            <div>
                                <p>
                                    <button onClick={handleGoBack}>go back</button>
                                    <button onClick={handleRedirect}>go home</button>
                                </p>
                            </div>
                            <h1>
                                {name}
                            </h1>
                            {army_type && <p>Army type: <span>{army_type}</span></p>}
                            {unique_unit && Boolean(unique_unit.length) &&
                                <div>
                                    <p>Unique unit:</p>
                                    <ul>
                                        {unique_unit.map((x, i) => <li key={i}>{x}</li>)}
                                    </ul>
                                </div>}
                            {unique_tech && Boolean(unique_tech.length) &&
                                <div>
                                    <p>Unique tech:</p>
                                    <ul>
                                        {unique_tech.map((x, i) => <li key={i}>{x}</li>)}
                                    </ul>
                                </div>}
                            {team_bonus && <p>Team bonus: <span>{team_bonus}</span></p>}
                            {description && <p>Description: <span>{description}</span></p>}
                            {expansion && <p>Expansion: <span>{expansion}</span></p>}
                            {age && <p>Age: <span>{age}</span></p>}
                            {created_in && <p>Created in: <span>{created_in}</span></p>}
                            {cost && <div><p>Costs:</p>
                                <ul>
                                    {cost.Wood && <li>Wood - <span>{cost.Wood}</span></li>}
                                    {cost.Gold && <li>Gold - <span>{cost.Gold}</span></li>}
                                    {cost.Food && <li>Food - <span>{cost.Food}</span></li>}
                                    {cost.info && <li>Cost - <span>{cost.info}</span></li>}
                                    {cost.Stone && <li>Stone - <span>{cost.Stone}</span></li>}

                                </ul>
                            </div>}
                            {build_time && <p>Build time: <span>{build_time}</span></p>}
                            {reload_time && <p>Reload time: <span>{reload_time}</span></p>}
                            {attack_delay && <p>Attack delay: <span>{attack_delay}</span></p>}
                            {movement_rate && <p>Movement rate: <span>{movement_rate}</span></p>}
                            {line_of_sight && <p>Line of sight: <span>{line_of_sight}</span></p>}
                            {hit_points && <p>Hit_points: <span>{hit_points}</span></p>}
                            {range && <p>Range: <span>{range}</span></p>}
                            {attack && <p>Attack: <span>{attack}</span></p>}
                            {armor && <p>Armor: <span>{armor}</span></p>}
                            {attack_bonus && <div><p>Attack bonus:</p>
                                <ul>
                                    {attack_bonus.map((x, i) => <li key={i}>{x}</li>)}
                                </ul>
                            </div>
                            }
                            {special &&
                                <div>
                                    <p>Special:</p>
                                    <ul>
                                        {special.map((x, i) => <li key={i}>{x}</li>)}
                                    </ul>
                                </div>
                            }
                            {civilization_bonus &&
                                <div>
                                    <p>Civilization bonus:</p>
                                    <ul>
                                        {civilization_bonus.map((x, i) => <li key={i}>{x}</li>)}
                                    </ul>
                                </div>
                            }
                            {applies_to &&
                                <div>
                                    <p>Applies to:</p>
                                    <ul>
                                        {applies_to.map((x, i) => <li key={i}>{x}</li>)}
                                    </ul>
                                </div>
                            }
                            {search_radius && <p>Search radius: <span>{search_radius}</span></p>}
                            {accuracy && <p>Accuracy: <span>{accuracy}</span></p>}

                        </div>
                }
            </div >)
    }
}
export default ItemsInfo