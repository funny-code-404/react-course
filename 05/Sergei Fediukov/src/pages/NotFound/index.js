const NotFound = () => {
    return (
        <div>
            <h1>
                page doesn't exist
           </h1>
        </div>
    )
}
export default NotFound