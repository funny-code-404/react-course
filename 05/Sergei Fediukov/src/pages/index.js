export { default as Home } from './Home'
export { default as Civilizations } from './Civilizations'
export { default as Units } from './Units'
export { default as Structures } from './Structures'
export { default as Technologies } from './Technologies'
export { default as NotFound } from './NotFound'

export { default as Item } from './Item'
export { default as UniqueUnit } from './UniqueUnit'
export { default as UniqueTech } from './UniqueTech'
export { default as UniqueCreatedIn } from './UniqueCreatedIn'
