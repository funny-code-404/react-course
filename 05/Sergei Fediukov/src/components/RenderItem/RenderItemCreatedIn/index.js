import { UniqueCreatedIn } from '../../../pages'
import React from 'react'
class RenderItemCreatedIn extends React.Component {
    state = {
        created_in: false,
    }
    render() {
        const created_in = this.props.created_in
        const hendleCreatedIn = () => {
            this.setState(prevState => ({
                ...prevState,
                created_in: !prevState.created_in
            }))
        }
        return (
            <div>
                <p>Created in:</p>
                <ul>
                    <li>
                        <button onClick={hendleCreatedIn}>
                            {
                                !this.state.created_in ?
                                    created_in.split('/')[created_in.split('/').length - 1].replace('_', ' ')
                                    : 'close'
                            }
                        </button>
                        {
                            this.state.created_in &&
                            <div>
                                <UniqueCreatedIn url={created_in} />
                            </div>
                        }
                    </li>
                </ul>
            </div>
        )
    }
}
export default RenderItemCreatedIn

