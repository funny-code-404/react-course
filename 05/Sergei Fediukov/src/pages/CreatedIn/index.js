import React from 'react'
import { connect } from 'react-redux'
import { getUniqueDetails } from '../../redux/actions';
import RenderItem from '../../components/RenderItem'

export class UniqueCreatedIn extends React.Component {

    componentDidMount() {
        const path = this.props.url.replace('https://age-of-empires-2-api.herokuapp.com/api/v1/', '/')
        this.props.getUniqueCreatedInDetails(path)
    }
    render() {
        return (
            <div>
                <RenderItem
                    data={this.props.uniqueCreatedIn.data}
                    unique='true' />
            </div >
        )
    }
}
const mapStateToProps = state => state;

const mapDispatchToProps = dispatch => ({
    getUniqueCreatedInDetails: path => dispatch(getUniqueDetails(path)),
});
export default connect(mapStateToProps, mapDispatchToProps)(UniqueCreatedIn) 
