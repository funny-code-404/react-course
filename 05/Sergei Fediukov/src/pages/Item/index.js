import React from 'react'
import { connect } from 'react-redux';
import { paths } from '../../config';
import { getDetails } from '../../redux/actions';
import RenderItem from '../../components/RenderItem'

class ItemsInfo extends React.Component {

    componentDidMount() {
        let path = ''
        if (this.props.location) {
            if (this.props.location.pathname.includes(paths.civilizations)) {
                path = paths.civilization
            }
            if (this.props.location.pathname.includes(paths.units)) {
                path = paths.unit
            }
            if (this.props.location.pathname.includes(paths.structures)) {
                path = paths.structure
            }
            if (this.props.location.pathname.includes(paths.technologies)) {
                path = paths.technology
            }
        }
        (+this.props.match.params.id) ?
            this.props.getItemDetails(`${path}/${this.props.match.params.id}`)
            : this.props.history.goBack()
    }
    render() {
        // console.log(this.props)
        return (

            <RenderItem data={this.props.item.data} history={this.props.history} />
        )
    }
}
const mapStateToProps = state => state;

const mapDispatchToProps = dispatch => ({
    getItemDetails: path => dispatch(getDetails(path)),
});
export default connect(mapStateToProps, mapDispatchToProps)(ItemsInfo)