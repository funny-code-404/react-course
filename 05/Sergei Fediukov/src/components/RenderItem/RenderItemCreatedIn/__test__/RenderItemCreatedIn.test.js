import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemCreatedIn from '..'
import { UniqueCreatedIn } from '../../../../pages'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {
    const mockFn = jest.fn();
    const state = {
        created_in: false,
    }
    const props = {
        created_in: '',
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<RenderItemCreatedIn {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component has children button', () => {
        expect(wrapper.find('button')).toHaveLength(1)
    })
    it('should change state when clicking', () => {
        wrapper.find('button').simulate('click')
        expect(wrapper.setState(prevState => ({
            ...prevState,
            created_in: !prevState.created_in
        })))
    })
    describe('should component UniqueCreatedIn appear correctly', () => {
        it('should change components`s state', () => {
            wrapper.setState({
                created_in: true,
            })
            expect(wrapper.find(UniqueCreatedIn)).toHaveLength(1)
        })
    })
})