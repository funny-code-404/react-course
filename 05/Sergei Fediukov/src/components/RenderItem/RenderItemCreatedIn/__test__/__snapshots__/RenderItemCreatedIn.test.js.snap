// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div>
  <p>
    Created in:
  </p>
  <ul>
    <li>
      <button
        onClick={[Function]}
      />
    </li>
  </ul>
</div>
`;
